db/src/main/java/io/lantern/db/SharedPreferencesAdapter.kt

```diff
@@ -28,38 +46,51 @@ class SharedPreferencesAdapter(
     }
 
     override fun getAll(): MutableMap<String, *> {
-        return HashMap(
-            db.list<Any>("%").map { it.path to it.value }
-                .toMap()
-        )
+        return HashMap(cache)
     }
 
     override fun getString(key: String, defValue: String?): String? {
-        return db.get(key) ?: defValue
+        return cache[key]?.let { it as String } ?: defValue
     }
 
     override fun getStringSet(key: String, defValues: MutableSet<String>?): MutableSet<String> {
         TODO("Not yet implemented")
     }
 
     override fun getInt(key: String, defValue: Int): Int {
-        return db.get(key) ?: defValue
+        var value = cache[key!!] ?: defValue
+        return when (value) {
+            is Number -> value.toInt()
+            is String -> value.toInt()
+            else -> throw ClassCastException("$value cannot be cast to Int")
+        }
     }
 
     override fun getLong(key: String, defValue: Long): Long {
-        return db.get(key) ?: defValue
+        var value = cache[key!!] ?: defValue
+        return when (value) {
+            is Number -> value.toLong()
+            is String -> value.toLong()
+            else -> throw ClassCastException("$value cannot be cast to Long")
+        }
     }
 
     override fun getFloat(key: String, defValue: Float): Float {
-        return db.get(key) ?: defValue
+        return cache[key]?.let { it as Float } ?: defValue
     }
 
     override fun getBoolean(key: String, defValue: Boolean): Boolean {
-        return db.get(key) ?: defValue
+        var value = cache[key!!] ?: defValue
+        return when (value) {
+            is Boolean -> value
+            is Number -> value.toInt() == 1
+            is String -> value.toBoolean()
+            else -> throw ClassCastException("$value cannot be cast to Boolean")
```

Comment:
wouldn't it make sense to return default value instead of throwing?

Reply:
Maybe :)  Right now the meaning of the default is "use this if no value is set". Returning it here would slightly alter that to also include "use this if we can't make sense of the value that's set".